### ISSUE TITLE
Support zod data transformation

### ISSUE THREAD

**Issue by Frantss** [fabian-hiller/modular-forms#20]
Hi, awesome library!

**Is your feature request related to a problem? Please describe.**
I would like the option of transforming my form data with the same zod validation schema

**Describe the solution you'd like**
```typescript
const schema = z.object({
  currency: z.string().length(3),
  amount: z.number().nonnegative().transform(amount => amount * 100),
});

const form = createForm(
  validate: zodForm(schema)
);

const onSubmit = data => {
    console.log(data.amount); // should be the user's input times 100
};
```

**Describe alternatives you've considered**
Passing a `transform` option to the schema validator, although IMO this should be an opt-out behavior since transforming data with zod is really common
```typescript
const form = createForm(
  validate: zodForm(schema, { transform: true })
);
```

Passing a transform function
```typescript
const form = createForm(
  validate: zodForm(schema)
  transform: zodTransform(schema)
);
```
Note that in this case `zodForm` could also be renamed to `zodValidate` to better reflect what it does here

**Additional context**
I'm currently able to achieve the same by just parsing the values again in the `onSubmit` which shouldn't fail since it was already validated with that same schema.

**Comment by fabian-hiller** [comment]
Thank you for your interest in Modular Forms and your commitment to improve the library. Already at the very beginning, before the integration of Zod, I thought about a transform feature. Primarily to transform data from the database, e.g. cent amounts to decimal numbers and then, before storing them in the database, back to cent amounts. Furthermore, a transform feature could be useful to bring phone numbers or IBANs into the correct formatting. However, the implementation must be well thought out so that the API remains simple and understandable. It is crucial that the user can decide at the field level when the transformation takes place.

In your case, until Modular Forms has this functionality, I would recommend you write a small helper function that does the transformation for you after the form is submitted. Below is an example.

```tsx
import { createForm, Form , zodForm} from '@modular-forms/solid';
import { z } from 'zod';

// utils/transformSubmit.ts

function transformSubmit<T extends z.ZodType<any, any, any>>(
  schema: T,
  action: (values: z.output<T>) => void | Promise<void>
) {
  return (values: z.input<T>) => action(schema.parse(values));
}

// routes/form.tsx

const schema = z.object({
  currency: z.string().length(3),
  amount: z
    .number()
    .nonnegative()
    .transform((amount) => amount * 100),
});

export default function FormRoute() {
  const form = createForm<z.input<typeof schema>>({
    validate: zodForm(schema),
  });

  const handleSubmit = transformSubmit(schema, (values) => {
    …
  });

  return (
    <Form of={form} onSubmit={handleSubmit}>
      …
    </Form>
  );
}
```

`zodForm` cannot be renamed to `zodValidate` because there is also a `zodField` adapter. By the current naming it is clear where the respective adapter must be applied. Also, if you look closely with `validate: zodForm(schema)` it leads to a readable and understandable API. However, I am open to improvements if you have more ideas on this.

**Comment by Frantss** [comment]
Thanks for the quick response! Will be using this helper in the mean time.

Regarding renaming `zodForm` and `zodField`, that seems only necessary if a zod transform function is introduced, in that case i could see something like `zodValidateForm`, `zodValideField`, `zodTransformForm`, and `zodTransformField`

**Comment by juanpmarin** [comment]
@fabian-hiller why don't allow transformations by default? I think that's an intuitive default behavior

**Comment by fabian-hiller** [comment]
This has to do with the history of Modular Forms. Initially, I didn't implement it for schema libraries. To support that, I would have to rewrite some code.

**Comment by juanpmarin** [comment]
@fabian-hiller thanks for the answer! 
Is there a way to apply transformations in the back side without parsing the object twice?

**Comment by fabian-hiller** [comment]
I don't think so. The problem is that `valiForm$` and `zodForm$` only return an error map and not the output of the schema. But there are plans to rewrite Modular Forms and optimize its code for schema libraries.
